code/components/citizen-server-impl/src/state/ServerSetters.cpp

```diff
@@ -447,6 +539,163 @@ static InitFunction initFunction([]()
 			ctx.SetResult(sgs->MakeScriptHandle(entity));
 		});
 
+		fx::ScriptEngine::RegisterNativeHandler("CREATE_TRAIN", [=](fx::ScriptContext& ctx)
+		{
+			uint32_t resourceHash = 0;
+
+			fx::OMPtr<IScriptRuntime> runtime;
+
+			if (FX_SUCCEEDED(fx::GetCurrentScriptRuntime(&runtime)))
+			{
+				fx::Resource* resource = reinterpret_cast<fx::Resource*>(runtime->GetParentObject());
+
+				if (resource)
+				{
+					resourceHash = HashString(resource->GetName().c_str());
+				}
+			}
+
+			uint32_t modelHash = ctx.GetArgument<uint32_t>(0);
+			float x = ctx.GetArgument<float>(1);
+			float y = ctx.GetArgument<float>(2);
+			float z = ctx.GetArgument<float>(3);
+			bool direction = ctx.GetArgument<bool>(4);
+			bool stopAtStations = ctx.GetArgument<bool>(5);
+			float speed = ctx.GetArgument<float>(6);
+			uint32_t trackId = ctx.GetArgument<uint32_t>(7);
+			uint32_t trainConfigIndex = ctx.CheckArgument<uint32_t>(8);
+
+			//Don't create train if an invalid trackid has been provided.
+			if (trackId != 0 && trackId != 3)
```

Comment:
While it's understandable to check for track IDs, it's possible for servers to load new tracks into the game. This information is not available to the server though, so the check (afaik) can't be changed to include those. Plus, the default tracks outside of ID 0 and 3 are perfectly valid and I see no reason why you would not be allowed to use them.

Reply:
Since it's not currently feasable to have the server be aware of valid track and train config IDs, then a way to bypass these checks would be useful, at least for servers that specifically have additional content. 
Possibly ConVar check to bypass, or some way to "add valid IDs"(?)